### PR TITLE
Bump python version to 3.6 so that we may use new features

### PR DESCRIPTION
**Pull request by vtomole**
Fixes: https://github.com/quantumlib/Cirq/issues/1693

### CODE REVIEW

**Comment by Strilanc** [comment]
Replace the meat of `_version.py` with this code, which will allow us to properly warn people about dropping versions:

```python
import sys

if sys.version_info < (3, 6, 0):
    raise SystemError(
        "You installed the latest version of cirq but aren't on python 3.6+.\n"
        'To fix this error you need to either:\n'
        '\n'
        'A) Update to python 3.6 or later.\n'
        '- OR -\n'
        'B) Explicitly install an older deprecated-but-compatible version '
        'of cirq (e.g. "python -m pip install cirq==0.5.*")')


__version__ = "0.6.0.dev"
```

**Comment by Strilanc** [comment]
The installation instructions need to be updated to talk about the new version.

**Comment by vtomole** [comment]
@PTAL

File: cirq/_version.py [hunk 1]
```diff
@@ -12,13 +12,14 @@
...
-"""Define version number here, read it from setup.py automatically, and warn users that
-the latest version of cirq uses python 3.6+"""
+"""Define version number here, read it from setup.py automatically,
+and warn users that the latest version of cirq uses python 3.6+"""
 
 import sys
 
 if sys.version_info < (3, 6, 0):
-    raise SystemError(
+    # TODO: Find the best way to cover this. Maybe mock sys.version_info
```
Comment (Strilanc) [inline cirq/_version.py]: This does not need to be covered. Just mark it `# coverage: ignore`.

Comment (vtomole) [reply cirq/_version.py]: Done.

**Pull request merged**